glusterd2/transaction/transaction.go

```diff
@@ -138,6 +140,9 @@ func (t *Txn) Do() error {
 				t.Ctx.Logger().WithError(err).Error("Transaction failed, rolling back changes")
 				t.undo(i)
 			}
+			if t.DontCheckAlive && isNodeUnreachable(err) {
```

Comment:
The earlier place was better. Checking it here will trigger undo. Isn't it ?

Reply:
yes, the earlier place was better, if an error is other than unreachable undo will get triggered.